### PR TITLE
coalib/bears/Bear.py: Fix warning when debug enabled

### LINKED ISSUE #3793: Warning says to enable debug when debug is already enabled
When running coala with `-V` , a warning message says to enable -V

```
Bear CheckstyleBear failed to run. Take a look at debug messages (`-V`) for further information.
```
https://ci.appveyor.com/project/coala/coala-bears/build/job/c9ca1wygut9y7eds

### PR DESCRIPTION
**Pull request by sourabh1031**
removed warning when debug mode is already enabled

Fixes #3793 

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [X] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [X] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by RaiVaibhav** [comment]
you still didn't get where to change the code, the main problem here is when user tries `-V` then also it is showing the warning message which we don't want. So I said you to check the log_level in section and if log_level is already set then it means that user is already in debug mode so you don't need to display the warning.

**Comment by RaiVaibhav** [comment]
@rishabhgarg25699 
>you did not understand the problem and you are changing the code in wrong place . You do not have to change the code of Bear.py You have to change the code of SpaceConsistencyBear.py in coala-bears to throw an exception at run()

seems like, you also didn't understand the problem, please understand how bear run, all the bear whether local or Global they get inherited from bear base class.

**Comment by RaiVaibhav** [comment]
@rishabhgarg25699  review your own review please.

File: coalib/bears/Bear.py [hunk 1]
```diff
@@ -461,13 +461,16 @@ def execute(self, *args, debug=False, **kwargs):
...
 
             if self.kind() == BEAR_KIND.LOCAL:
-                self.err('Bear {} failed to run on file {}. Take a look '
-                         'at debug messages (`-V`) for further '
-                         'information.'.format(name, args[0]))
+                if self.section['log_level'].value == 'DEBUG':
+                    pass
+                else:
+                    self.err('Bear {} failed to run on file {}. Take a look '
+                        'at debug messages (`-V`) for further '
```
[the remark this answers is not in the document]
Comment (sladyn98) [reply coalib/bears/Bear.py]: The PEP guideline is telling you to indent the lines at the start of the bracket, which is one of the pep requirements.

[the remark this answers is not in the document]
Comment (sourabh1031) [reply coalib/bears/Bear.py]: @sladyn98 i have already fixed up those issues

[the remark this answers is not in the document]
Comment (sourabh1031) [reply coalib/bears/Bear.py]: @sladyn98 and please review only after you have seen what is to be reviewed as my test case for automated pr and commit checks are passed by gitmate

**Comment by sladyn98** [comment]
Approach the problem in tasks.
First check the log level and print and then proceed. The output is relayed to the log_printer, be sure to know what is happening and you will easily solve it.

File: coalib/bears/Bear.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (RaiVaibhav) [reply coalib/bears/Bear.py]: This is not a way to talk to anyone, he is giving his time on review of your PR, anyone can make a mistake, in the current scenario you can simply say that `you have already resolved the review by gitmate`

[the remark this answers is not in the document]
Comment (sladyn98) [reply coalib/bears/Bear.py]: I am so sorry, if you need my help in any issue I would love to assist you.

File: coalib/bears/Bear.py [hunk 2]
```diff
@@ -461,13 +461,18 @@ def execute(self, *args, debug=False, **kwargs):
                          .format(name), str(exc))
 
             if self.kind() == BEAR_KIND.LOCAL:
```
Comment (RaiVaibhav) [inline coalib/bears/Bear.py]: you can add an inline logic here instead of adding extra code.

File: coalib/bears/Bear.py [hunk 3]
```diff
@@ -461,13 +461,18 @@ def execute(self, *args, debug=False, **kwargs):
...
-                         'information.'.format(name, args[0]))
+                if self.section['log_level'].value == 'DEBUG':
+                    pass
+                else:
+                    self.err(
+                        'Bear {} failed to run on file {}. Take a look '
+                        'at debug messages (`-V`) for further '
+                        'information.'.format(name, args[0]))
             else:
-                self.err('Bear {} failed to run. Take a look '
```
Comment (RaiVaibhav) [inline coalib/bears/Bear.py]: Unrelated change, revert it back.

File: coalib/bears/Bear.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sourabh1031) [reply coalib/bears/Bear.py]: @sladyn98 i am really sorry as i hurted your emotions and hard work but trust me i was not angry at you.surely you can review my pr no problem i would appreciate to learn from you

[the remark this answers is not in the document]
Comment (sourabh1031) [reply coalib/bears/Bear.py]: and sorry to you too Vaibhav as i  didn't follow communication etiquettes

File: coalib/bears/Bear.py (hunk 2 above)
Comment (sourabh1031) [reply coalib/bears/Bear.py]: but adding inline logic will increase the length of line to be more than 80 chars

Comment (li-boxuan) [reply coalib/bears/Bear.py]:
```python
if self.kind() == BEAR_KIND.LOCAL and self.section['log_level'].value != 'DEBUG':
```

Comment (li-boxuan) [reply coalib/bears/Bear.py]: The benefit is that you don't need to do changes unrelated to the real problem.

Comment (sourabh1031) [reply coalib/bears/Bear.py]: no problem @li-boxuan you are helping me that is in fact great thanks

File: coalib/bears/Bear.py [hunk 4]
```diff
@@ -460,14 +460,16 @@ def execute(self, *args, debug=False, **kwargs):
                 self.err('Bear {} violated one-based offset convention.'
                          .format(name), str(exc))
 
-            if self.kind() == BEAR_KIND.LOCAL:
-                self.err('Bear {} failed to run on file {}. Take a look '
-                         'at debug messages (`-V`) for further '
-                         'information.'.format(name, args[0]))
+            if self.kind() == BEAR_KIND.LOCAL and self.section['log_level'].value != 'DEBUG':
```
[the remark this answers is not in the document]
Comment (sourabh1031) [reply coalib/bears/Bear.py]: resolved

[the remark this answers is not in the document]
Comment (sourabh1031) [reply coalib/bears/Bear.py]: resolved

**Comment by akshatkarani** [comment]
Commit message is wrong. Use imperative present tense.

**Comment by sourabh1031** [comment]
> Commit message is wrong. Use imperative present tense.

Fixes warning when debug enabled <--- what is wrong in this please tell

**Comment by sladyn98** [comment]
> > Commit message is wrong. Use imperative present tense.
> 
> Fixes warning when debug enabled <--- what is wrong in this please tell it uses imperative present only

"Remove" is the keyword to be used.

**Comment by sourabh1031** [comment]
> > > Commit message is wrong. Use imperative present tense.
> > 
> > 
> > Fixes warning when debug enabled <--- what is wrong in this please tell it uses imperative present only
> 
> "Remove" is the keyword to be used.

ahh that is right thanks

**Comment by akshatkarani** [comment]
> > Commit message is wrong. Use imperative present tense.
> 
> Fixes warning when debug enabled <--- what is wrong in this please tell it uses imperative present only
I guess it should be fix. Also you have not given issue reference

**Comment by sourabh1031** [comment]
> > > Commit message is wrong. Use imperative present tense.
> > 
> > 
> > Fixes warning when debug enabled <--- what is wrong in this please tell it uses imperative present only
> > I guess it should be fix. Also you have not given issue reference

actually i have written here 
fixes #3793 or should i mention somewhere else as well

**Comment by RaiVaibhav** [comment]
Google search commit message

File: coalib/bears/Bear.py [hunk 5]
```diff
@@ -460,14 +460,17 @@ def execute(self, *args, debug=False, **kwargs):
                 self.err('Bear {} violated one-based offset convention.'
                          .format(name), str(exc))
 
-            if self.kind() == BEAR_KIND.LOCAL:
-                self.err('Bear {} failed to run on file {}. Take a look '
-                         'at debug messages (`-V`) for further '
-                         'information.'.format(name, args[0]))
+            if self.kind() == BEAR_KIND.LOCAL and
+                self.section['log_level'].value != 'DEBUG':
+                self.err(
```
[the remark this answers is not in the document]
Comment (sourabh1031) [reply coalib/bears/Bear.py]: done

File: coalib/bears/Bear.py [hunk 6]
```diff
@@ -460,14 +460,17 @@ def execute(self, *args, debug=False, **kwargs):
                 self.err('Bear {} violated one-based offset convention.'
                          .format(name), str(exc))
 
-            if self.kind() == BEAR_KIND.LOCAL:
-                self.err('Bear {} failed to run on file {}. Take a look '
-                         'at debug messages (`-V`) for further '
-                         'information.'.format(name, args[0]))
+            if self.kind() == BEAR_KIND.LOCAL
```
Comment (Man-Jain) [inline coalib/bears/Bear.py]: Ci is showing syntax error on this line So I think the way you took the and part on next line is messibg the syntax. You'll need to fix that before anything else.

Comment (sourabh1031) [reply coalib/bears/Bear.py]: @Man-Jain right I too looked and found the same

File: coalib/bears/Bear.py [hunk 7]
```diff
@@ -460,14 +460,17 @@ def execute(self, *args, debug=False, **kwargs):
...
+            and self.section['log_level'].value != 'DEBUG':
```
Comment (Man-Jain) [inline coalib/bears/Bear.py]: Also if we think about your logic it says
If the Debug is not set then print error else print error. So its gonna print the error in both the situations.

File: coalib/bears/Bear.py (hunk 6 above)
Comment (sourabh1031) [reply coalib/bears/Bear.py]: but actually if i am writting on a single line then problem and if i am indenting the remaining part on next line then problem what should i do ?

Comment (Man-Jain) [reply coalib/bears/Bear.py]: Put the conditions in brackets.

File: coalib/bears/Bear.py (hunk 3 above)
Comment (li-boxuan) [reply coalib/bears/Bear.py]: The logic is not correct here. Your code is saything that If `self.kind() == BEAR_KIND.LOCAL` and `self.section['log_level'].value == 'DEBUG'`, display error message for local bear. Otherwise, display error message for global bear.

Comment (sourabh1031) [reply coalib/bears/Bear.py]: @li-boxuan @RaiVaibhav  i had already updated my pr but you haven't reviewed please review

**Comment by li-boxuan** [comment]
ping @sourabh1031

**Comment by sourabh1031** [comment]
> ping @sourabh1031

@li-boxuan yes please :)

**Comment by li-boxuan** [comment]
@sourabh1031 I am pinging you because this pr has been stale for 10 days ;) You need to improve your PR and let CI pass.

File: coalib/bears/Bear.py [hunk 8]
```diff
@@ -460,14 +460,17 @@ def execute(self, *args, debug=False, **kwargs):
                 self.err('Bear {} violated one-based offset convention.'
                          .format(name), str(exc))
 
-            if self.kind() == BEAR_KIND.LOCAL:
-                self.err('Bear {} failed to run on file {}. Take a look '
-                         'at debug messages (`-V`) for further '
-                         'information.'.format(name, args[0]))
+            if (self.kind() == BEAR_KIND.LOCAL
+                    or self.section['log_level'].value == 'DEBUG'):
+                self.err(
```
Comment (jayvdb) [inline coalib/bears/Bear.py]: As @RaiVaibhav said at https://github.com/coala/coala/pull/5772/files#r223205682 , you have made changes in your PR which are meaningless.  Here you do

```diff
-self.err('Bear {} failed to run on file {}. Take a look '
+self.err(
+    'Bear {} failed to run on file {}. Take a look '
```

dont do that.

Dont touch other lines if you do not need to modify them.

It makes your patch bigger and harder to read, and therefore harder to review.

And your PR will be marked as a WIP because of this, because you havent yet provided a neat PR for review.

File: coalib/bears/Bear.py [hunk 9]
```diff
@@ -460,7 +460,8 @@ def execute(self, *args, debug=False, **kwargs):
                 self.err('Bear {} violated one-based offset convention.'
                          .format(name), str(exc))
 
-            if self.kind() == BEAR_KIND.LOCAL:
+            if (self.kind() == BEAR_KIND.LOCAL
+                    or self.section['log_level'].value == 'DEBUG'):
```
Comment (jayvdb) [inline coalib/bears/Bear.py]: this logic does not achieve what the issue requested.

**Comment by sourabh1031** [comment]
@jayvdb @li-boxuan but the mentor guided me this way and i don't know why still the logic is not correct.can you tell the approach in which i should think to this problem

**Comment by jayvdb** [comment]
Read https://github.com/coala/coala/pull/5772/files

**Comment by jayvdb** [comment]
> but the mentor guided me this way

Which mentor?  Specifically , which comment -- provide a URL to the comment you are referring to.

**Comment by sourabh1031** [comment]
@jayvdb Vaibhav Rai before sending pr i asked whether whatever logic i am pushing is correct or not

**Comment by jayvdb** [comment]
I am not blame him @RaiVaibhav  either.  The logic you started with was better than the current logic.  Then it was changed.  I know it is now not correct.  That is all that matters.  And it is your task to write the correct logic, based on the problem described on the issue

**Comment by RaiVaibhav** [comment]
@sourabh1031 I never said your logic is correct, you were trying hit and trail method previously so I said you to open the PR instead of asking the same thing everytime and pinging me everytime, see me review comment on the top https://github.com/coala/coala/pull/5772#issuecomment-425058040

**Comment by RaiVaibhav** [comment]
@sourabh1031 I am not a mentor, you asked the doubt on channel I helped there, you were sending the pastebin code everytime thats why I said you to open a PR and do all the changea on that.

**Comment by sourabh1031** [comment]
and one more thing, if either of you can help me here, is the following two commands works the same
coala --no-config --files=sample.txt --bears=SpaceConsistencyBear -V
coala --no-config -f sample.txt -b SpaceConsistencyBear -V
Actually i ran the both commands along with --debug-bears and in first one i was not able to find the log_level key while in the second one i was. for more reference here is the pastebin link:
https://pastebin.com/btNAn4ib

**Comment by RaiVaibhav** [comment]
@sourabh1031 check your first command on pastebin you didn't use the `-V`, please check it before comment all the reviewer gets notified unnecessary.
